0000-Install-and-Test/NodeRED-and-TJBot-Nodes/soict_s4v/tests/node_modules/tjbot/lib/tjbot.js

```diff
@@ -0,0 +1,1552 @@
+/**
+ * Copyright 2016-2018 IBM Corp. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+// useful node modules
+const assert = require('assert');
+const temp = require('temp').track();
+const Promise = require('bluebird');
+const fs = require('fs');
+const sleep = require('sleep');
+const colorToHex = require('colornames');
+const cm = require('color-model');
+const winston = require('winston');
+const {exec} = require("child_process");  //TienND: for dummy led NeoPixel function
+
+// hardware modules
+const Mic = require('mic');
+const Raspistill = require('node-raspistill').Raspistill;
+
+/**
+ * TJBot
+ * @param {String} hardware The set of hardware with which TJBot is equipped (see TJBot.prototype.hardware).
+ * @param {Object} configuration Configuration parameters
+ * @param {Object} credentials The set of service credentials needed for external services (see TJBot.prototype.services).
+ * @constructor
+ */
+function TJBot(hardware, configuration, credentials) {
+    if (!(this instanceof TJBot)) {
+        throw new Error('"new" keyword required to create TJBot service instances');
+    }
+
+    // import configuration params
+    this.configuration = Object.assign({}, TJBot.prototype.defaultConfiguration, configuration);
+
+    // set logging level
+    winston.level = this.configuration.log.level;
+
+    // set up the hardware
+    if (hardware == undefined) {
+        throw new Error('must define a hardware configuration for TJBot');
+    }
+
+    hardware.forEach(function(device) {
+        switch (device) {
+        case 'camera':
+            this._setupCamera();
+            break;
+
+        case 'led':
+            //this._setupLED();
```

Comment:
Why did you disable this line?

Reply:
This function is used to check the Pi version and change the configuration respectively.  Enabling this could makes the feature LED stop immediately.